Extension/src/packageManager.ts

```diff
@@ -160,25 +161,23 @@ export class PackageManager {
     }
 
     private ArchitecturesMatch(value: IPackage): boolean {
-        return !value.architectures || value.architectures.indexOf(this.platformInfo.architecture) !== -1;
+        return !value.architectures || (this.platformInfo.architecture !== undefined && value.architectures.indexOf(this.platformInfo.architecture) !== -1);
```

Comment:
Is there a reason to test against undefined specifically, or can you remove the `!== undefined` part?

Reply:
Like referred to above, in rateLimitExceeded(), testing the potentially null/undefined value is not compatible with boolean operators, as it does not result in a boolean value.  This is actually a cleaner approach (to use the !== undefined to convert it to a boolean), so I will use this in rateLimitExceeded().